packages/websocket/src/abstract-websocket.client.ts

```diff
@@ -96,26 +100,30 @@ export abstract class AbstractWebsocketClient<K> implements WebsocketHandler<K>,
         this.disconnect();
     }
 
+    protected serialize(message: Any): WebSocketMessage {
+        return JSON.stringify(message);
+    }
+
+    protected deserialize(messageEvent: MessageEvent): WebsocketMessage<K, Any> {
+        return JSON.parse(messageEvent.data);
```

Comment:
What do you think about?

![image](https://user-images.githubusercontent.com/12021443/117258316-80208480-ae55-11eb-9b8a-9a2ec22f9313.png)

Reply:
I can't imagine a case when `undefined` will get there. But even if your connection can return this, you can override these methods.

Reply:
case

```
// server send `ping` instead `{ data: null, result: null }`

// your code
// when messageEvent is "ping" -> "ping".data =>> undefined
protected deserialize(messageEvent: MessageEvent): WebsocketMessage<K, Any> {
        return JSON.parse(messageEvent.data);
}
```

Reply:
perhaps we should abandon the concept of `MessageEvent`, what do you think?

Reply:
MessageEvent is a JS Event, it carries a lot of metadata among the data. if the message is `ping`, the `messageEvent` will be `{ ..., data: 'ping' }`. 
i guess you mean `WebsocketMessage<K, T>`. i agree, `WebsocketMessage<K, T>` is something weird. but abandon involves a bunch of fixes, no need for this at this moment it think.

MessageEvent is a JS event, it carries a lot of metadata among `data`. if the message is "ping", `messageEvent` will be `{ ..., data: 'ping' }`.
I think you mean `WebsocketMessage <K, T>`. I agree, `WebsocketMessage <K, T>` is weird. but the refusal suggests a bunch of fixes, this is not necessary at the moment I think.

Reply:
okey